ethsigner/core/src/test/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/transaction/GoQuorumEthTransactionTest.java

```diff
@@ -0,0 +1,138 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.core.requesthandler.sendtransaction.transaction;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.web3j.utils.Bytes.trimLeadingZeroes;
+
+import tech.pegasys.ethsigner.core.jsonrpc.EeaSendTransactionJsonParameters;
+import tech.pegasys.ethsigner.core.jsonrpc.EthSendTransactionJsonParameters;
+import tech.pegasys.ethsigner.core.jsonrpc.JsonRpcRequest;
+import tech.pegasys.ethsigner.core.jsonrpc.JsonRpcRequestId;
+import tech.pegasys.ethsigner.core.requesthandler.sendtransaction.EnclaveLookupIdProvider;
+
+import java.math.BigInteger;
+import java.util.List;
+
+import io.vertx.core.json.JsonObject;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.web3j.crypto.Sign.SignatureData;
+import org.web3j.protocol.eea.crypto.PrivateTransactionDecoder;
+import org.web3j.protocol.eea.crypto.SignedRawPrivateTransaction;
+import org.web3j.utils.Base64String;
+import org.web3j.utils.Numeric;
+import org.web3j.utils.Restriction;
+
+public class GoQuorumEthTransactionTest {
+
+  private GoQuorumPrivateTransaction ethTransaction;
+  private EthSendTransactionJsonParameters params;
+
+  @BeforeEach
+  public void setup() {
+    params = new EthSendTransactionJsonParameters("0x7577919ae5df4941180eac211965f275cdce314d");
+    params.receiver("0xd46e8dd67c5d32be8058bb8eb970870f07244567");
+    params.gas("0x76c0");
+    params.gasPrice("0x9184e72a000");
+    params.nonce("0xe04d296d2460cfb8472af2c5fd05b5a214109c25688d3704aed5484f9a7792f2");
+    params.value("0x0");
+    params.data(
+        "0xd46e8dd67c5d32be8d46e8dd67c5d32be8058bb8eb970870f072445675058bb8eb970870f072445675");
+    // These extra attributes make it a GoQuorum private transaction via eth_sendTransaction
+    params.privateFrom("ZlapEsl9qDLPy/e88+/6yvCUEVIvH83y0N4A6wHuKXI=");
+    params.privateFor(new String[] {"GV8m0VZAccYGAAYMBuYQtKEj0XtpXeaw2APcoBmtA2w="});
+
+    EnclaveLookupIdProvider provider =
+        (x) ->
+            "9alPvwI5WX9Ct/1DUdNSvCdhj0bLvw+f7NZ/1oG9IaznAspXyAlqp30YzKHcx8oe+QBrnrKPldoPzy98bA7ABg==";
+    ethTransaction =
+        GoQuorumPrivateTransaction.from(
+            params, () -> BigInteger.ZERO, provider, new JsonRpcRequestId(1));
+  }
+
+  @Test
+  public void eeaSendTxParamsFromEthSendTxParams() {
+    EeaSendTransactionJsonParameters eeaParams = new EeaSendTransactionJsonParameters(params);
```

Comment:
I still don't like that we are reusing the Eea class. If there is a lot of duplication, could we just create an abstract class, or could we rename the class?

Reply:
gone